Sources/SwiftLexicalLookup/ScopeSyntax.swift

```diff
@@ -0,0 +1,106 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+extension SyntaxProtocol {
+  /// Returns all names that `for` refers to at this syntax node.
+  /// Optional configuration can be passed as `with` to customize the lookup behavior.
+  ///
+  /// - Returns: An array of `LookupResult` for name `for` at this syntax node,
+  /// ordered by visibility. If set to `nil`, returns all available names ordered by visibility.
+  /// The order is from the innermost to the outermost scope,
+  /// and within each result, names are ordered by their introduction
+  /// in the source code.
+  ///
+  /// Example usage:
+  /// ```swift
+  /// class C {
+  ///   var a = 42
+  ///
+  ///   func a(a: Int) {
+  ///     a // <--- lookup here
+  ///
+  ///     let a = 0
+  ///   }
+  ///
+  ///   func a() {
+  ///     // ...
+  ///   }
+  /// }
+  /// ```
+  /// When calling this function on the declaration reference `a` within its name,
+  /// the function returns the parameter first, then the identifier of the variable
+  /// declaration, followed by the first function name, and then the second function name,
+  /// in this exact order. The constant declaration within the function body is omitted
+  /// due to the ordering rules that prioritize visibility within the function body.
+  @_spi(Experimental) public func lookup(for name: String?, with configArr: [LookupConfig] = []) -> [LookupResult] {
```

Comment:
I think the array of `LookupConfig` is more abstract than we need. Any kind of configuration is going to need custom logic within the `lookup` function. I recommend replacing this array (and the `LookupConfig` protocol) with a simple struct with configuration options, e.g.,

```swift
struct LookupConfig {
  var fileScopeHandling: FileScopeHandlingConfig = (whatever the default is)
}
```

Over time, we can expand it if we need to. For example, we could add an optional build configuration when we're ready to deal with `#if`s:

```swift
  var buildConfiguration: (any BuildConfiguration)? = nil
```

Reply:
I introduced `LookupConfig` struct to simplify configuration handling. Thank you for pointing out using build configuration. Do you think we could also use it now to provide default file scope behavior? Or should we leave it for later when we have more configurations?

Reply:
I think it's fine to use it now